cwms_radar_api/src/main/java/cwms/radar/data/dao/JooqDao.java

```diff
@@ -80,4 +94,89 @@ public void accept(org.jooq.Context<?> ctx) {
         };
     }
 
+    // Not really wrapping or unwrapping
+    public static RuntimeException wrapException(RuntimeException input) {
+        RuntimeException retval = input;
+
+        if (isNotFound(input)) {
+            retval = buildNotFound(input);
+        }
+
+        return retval;
+    }
+
+    public static boolean isNotFound(RuntimeException input) {
+        boolean retval = false;
+
+        Throwable cause = input;
+        if (input instanceof DataAccessException) {
+            DataAccessException dae = (DataAccessException) input;
+            cause = dae.getCause();
+        }
+
+        if (cause instanceof SQLException) {
+            SQLException sqlException = (SQLException) cause;
+            final String localizedMessage = cause.getLocalizedMessage();
+
+            // See link for a more complete list of CWMS Error codes:
+            // https://bitbucket.hecdev.net/projects/CWMS/repos/cwms_database_origin_teamcity_work/browse/src/buildSqlScripts.py#4866
+
+            int errorCode = sqlException.getErrorCode();
+            if (errorCode == 20001 || errorCode == 20025 || errorCode == 20034
+                    || localizedMessage.contains("_DOES_NOT_EXIST")
+                    || localizedMessage.contains("_NOT_FOUND")
+                    || localizedMessage.contains(" does not exist.")
+            ) {
+                retval = true;
+            }
+        }
+        return retval;
+    }
+
+    @NotNull
+    static NotFoundException buildNotFound(RuntimeException input) {
+        // The cause can be kinda long and include all the line numbers from the pl/sql.
+
+        Throwable cause = input;
+        if (input instanceof DataAccessException) {
+            DataAccessException dae = (DataAccessException) input;
+            cause = dae.getCause();
+        }
+
+        NotFoundException exception = new NotFoundException(cause);
+
+        String localizedMessage = cause.getLocalizedMessage();
+        if (localizedMessage != null) {
+            String[] parts = localizedMessage.split("\n");
+            if (parts.length > 1) {
+                exception = new NotFoundException(parts[0]);
+            }
+        }
+        return exception;
+    }
+
+
+
+    private static class ExceptionWrappingListener extends DefaultExecuteListener {
```

Comment:
styleguide nitpick: inner classes at the bottom of the file

Reply:
I'd say this PR is currently only half-baked.  Hopefully discussion in #299 will lead to a more complete approach.